### PR TITLE
Add log-linear interpolation and refactor linear interp

### PR DESCRIPTION
**Pull request by wegamekinglc**
## Summary
- Extract linear interpolation into separate files (`interplinear.cpp`/`.hpp`) from the monolithic `interp.cpp`/`.hpp`
- Add new log-linear interpolation (`interploglinear.cpp`/`.hpp`) with corresponding auto-generated code
- Add auto-generated Machinist code for LogLinear1 object (`MG_LogLinear1_Read.inc`, `MG_LogLinear1_Write.inc`, `MG_LogLinear1_object.hpp`)
- Add unit tests for log-linear interpolation, including edge-case tests (non-positive f, non-monotonic x, mismatched sizes)
- Update all includes across the codebase to reference the new file locations
- Fix missing `algorithms.hpp` include in `interplinear.hpp`
- Fix stray backslash and duplicate include in `test_dictionary.cpp`
- Stop deleting build directory in build scripts for incremental builds
- Fix CLAUDE.md heading and unit-test-style include order
- Update .gitignore and unit-test skill description

## Test plan
- [x] Run `bin/test_suite --gtest_filter=InterpTest.*` to verify new log-linear interpolation tests and existing interpolation tests still pass
- [x] Full `bin/test_suite` to confirm no regressions

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/math/interp/test_interploglinear.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    // exact at knots
+    for (int i = 0; i < x.size(); ++i)
+        ASSERT_NEAR((*interp)(x[i]), f[i], 1e-10);
+
+    // mid-point: exp of linear interp of logs
+    double mid = 0.5;
+    double expected = std::exp(0.5 * std::log(1.0) + 0.5 * std::log(2.0));
+    ASSERT_NEAR((*interp)(mid), expected, 1e-10);
+}
```
[the remark this answers is not in the document]
Comment (wegamekinglc) [reply tests/math/interp/test_interploglinear.cpp:25]: agree
